test/cql-pytest/test_guardrail_replication_strategy.py

```diff
@@ -83,7 +83,7 @@ def test_given_already_existing_ks_when_altering_ks_should_validate_against_disc
             config_value_context(cql, 'replication_strategy_fail_list', 'EverywhereStrategy'))
 
         # create a ks with "allowed" strategy
-        with new_test_keyspace(cql, " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "' : 3 }") as keyspace:
+        with new_test_keyspace(cql, " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "' : 3 } AND TABLETS = {'enabled': false}") as keyspace:
```

Comment:
This will fail if run against an older version of Scylla. My PR handles this by only disabling tablets if they are enabled in the first place.

Reply:
It's not super important to be able to run all tests against older versions of Scylla, but I agree it's nicer if we can. I didn't like the fact you needed to add options to new_test_keyspace and magic keyspace names and so on. This is one of the things I least like about dtest's utility functions which each has a dozen options that try to "simplify" and in reality obscure the usage of the function: In this case new_test_keyspace just takes the entire keyspace definition as on CQL string, which test readers will immediately understand, instead of taking 7 different parameters and building a CQL string from it.

Reply:
There are no magic names, not sure what you are talking about. I added an option, so I don't have to repeat the checks at each call-site (disable tablets only if the scylla version supports it in the first place).